### PR TITLE
Remove some pushd popd, using git --git-dir --work-tree

### PR DESCRIPTION
**Pull request by jeremiejig**
Hey there,

Just moved some commit from #52 to a new PR.

### CODE REVIEW

**Comment by jeremiejig** [comment]
Mmmh, there are somes issue. fix pending.

**Comment by jeremiejig** [comment]
fix pushed, omf.repo.pull now need one argument.

**Comment by derekstavis** [comment]
Nice. You mind squashing into one commit?

**Comment by jeremiejig** [comment]
Well I'd like to keep the trace of the two solutions has one is more elegant `-C $git_path` and the other more git backward-compatible.

I'll squash the argument mandatory.

File: pkg/omf/cli/omf.repo.pull.fish [hunk 1]
```diff
@@ -1,42 +1,48 @@
 function omf.repo.pull
-  if test (command git config --get remote.upstream.url)
+  if test (count $argv) -eq 0
```
Comment (bpinto) [inline pkg/omf/cli/omf.repo.pull.fish]: As this is a function used only by `omf` I wouldn't bother adding a check for arguments. What do you think about this?

**Comment by derekstavis** [comment]
Worked like a charm on Linux!! :+1:

**Comment by bpinto** [comment]
Another great PR @jeremiejig ! thanks

**Pull request merged**